types/react-relay/index.d.ts

```diff
@@ -30,16 +26,19 @@ import {
 // ./ReactRelayTypes
 export interface RelayProp {
     environment: Environment;
+    refetch: undefined; // ensures no RelayRefetchProp is used with a fragment container
+    hasMore: undefined; // ensures no RelayPaginationProp is used with a fragment container
```

Comment:
Is this more appropriate than

```suggestion
    hasMore?: never; // ensures no RelayPaginationProp is used with a fragment container
```

Also, it seems like these should be optional regardless.

Reply:
Pretty sure I tried that and many other permutations. The tests would only pass with non-optional `undefined`.